lib/crypto.js

```diff
@@ -518,9 +519,11 @@ function ECDH(curve) {
   this._handle = new binding.ECDH(curve);
 }
 
-exports.createECDH = function createECDH(curve) {
-  return new ECDH(curve);
-};
+if (hasEC()) {
+  exports.createECDH = function createECDH(curve) {
+    return new ECDH(curve);
+  };
+}
```

Comment:
Yeah, I think throwing would be good solution. I still have to sort out some things on my end about the motivation for creating this in the first place, but throwing would def help with that issue too.

I'll take a shot at implementing this tomorrow. Thanks @bnoordhuis, @jasnell, and @sam-github  for your feedback on this.